test/apps/ruby-compile-env/Rakefile

```diff
@@ -0,0 +1,6 @@
+task 'assets:precompile' do
+  if ENV['FOO'] != 'BAR'
```

Comment:
You could just use the `env-dir` test app which prints the value of `FOO` on compile (see [here](https://github.com/flynn/flynn/blob/master/test/apps/env-dir/bin/compile#L10)), and then test `flynn-receiver -e FOO=BAR` prints `BAR` (you will need to use the inline buildpack like [this test](https://github.com/flynn/flynn/blob/master/test/test_git_deploy.go#L38)).